kythe/go/util/riegeli/reader.go

```diff
@@ -22,36 +22,73 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+
+	"github.com/golang/protobuf/proto"
+
+	rmpb "kythe.io/third_party/riegeli/records_metadata_go_proto"
 )
 
 func (r *Reader) nextRecord() ([]byte, error) {
-	if err := r.readNextRecordChunk(); err != nil {
+	if err := r.ensureRecordReader(); err != nil {
 		return nil, err
 	}
 	return r.recordReader.Next()
 }
 
-func (r *Reader) readNextRecordChunk() error {
+func (r *Reader) ensureRecordReader() error {
 	if r.recordReader != nil && r.recordReader.Len() == 0 {
 		if err := r.recordReader.Close(); err != nil {
 			return fmt.Errorf("error closing record reader: %v", err)
 		}
 		r.recordReader = nil
 	}
+
 	for r.recordReader == nil {
 		c, err := r.r.Next()
 		if err != nil {
 			return err
-		} else if c.Header.NumRecords == 0 {
-			// Skip all chunks with 0 records (even if chunk_type is unknown)
-			continue
-		} else if c.Header.ChunkType != recordChunkType {
-			// TODO(schroederc): support non-record chunk types (especially RecordsMetadata)
+		} else if c.Header.NumRecords == 0 && c.Header.ChunkType != fileMetadataChunkType {
+			// ignore chunks with no records; even for unknown chunk types
 			continue
 		}
-		r.recordReader, err = newRecordChunkReader(c)
-		if err != nil {
-			return fmt.Errorf("bad record chunk: %v", err)
+
+		switch c.Header.ChunkType {
+		case fileSignatureChunkType:
+			// TODO (schroederc): verify
```

Comment:
What does ignoring this do? Will the results be valid if we skip the signatures?

Reply:
Ignoring it means we could be reading a corrupt file, but skipping it won't be a problem for a valid file.

Reply:
It might be worth logging that. Although, if you're going to implement this soon anyway, it's not crucial.

Reply:
Better to just fix it.